packages/manager/apps/pci-block-storage/src/pages/detach/DetachStorage.page.tsx

```diff
@@ -11,14 +11,144 @@ import {
+import { TVolume } from '@/api/data/volume';
+
+const SelectInstance = ({ instances }: { instances: TAttachedInstance[] }) => {
+  const { control } = useFormContext();
+  return (
+    <div>
+      <Controller
+        control={control}
+        render={({ field: { value, onChange, onBlur, ...field } }) => (
+          <OsdsSelect
+            value={value}
+            {...field}
+            inline
+            className="mt-5 w-full"
+            onOdsBlur={onBlur}
+            onOdsValueChange={(event) => {
+              // Ods select sends value change event when incoming value has changed so we need to break it to avoid a call loop
+              if (event.detail.value !== value) onChange(event.detail.value);
+            }}
+            data-testid="detachStorage-select-instance"
+          >
+            {instances.map(({ id, name }) => (
+              <OsdsSelectOption key={id} value={id}>
+                {name}
+              </OsdsSelectOption>
+            ))}
+          </OsdsSelect>
+        )}
+        name="instanceId"
+      />
+    </div>
+  );
+};
+
+const InstanceDescription = ({
+  volume,
+  instances,
+}: {
+  volume: TVolume;
+  instances: TAttachedInstance[];
+}) => {
+  const { t } = useTranslation('detach');
+  const { control } = useFormContext();
+  const selectedInstanceId = useWatch({ control, name: 'instanceId' });
+
+  const selectedInstance = useMemo(() => {
+    if (selectedInstanceId)
+      return instances.find(({ id }) => id === selectedInstanceId) || null;
+    return null;
+  }, [selectedInstanceId, instances]);
+
+  if (!selectedInstance) return null;
+
+  return (
+    <div className="mt-5">
+      <OsdsText
+        level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+        size={ODS_THEME_TYPOGRAPHY_SIZE._400}
+        color={ODS_THEME_COLOR_INTENT.text}
+      >
+        {t('pci_projects_project_storages_blocks_block_detach_detachvolume', {
+          volume: volume.name,
+          instance: selectedInstance.name,
+        })}
+      </OsdsText>
+    </div>
+  );
+};
+
+const Form = <Schema extends z.ZodObject<ZodRawShape>>({
+  schema,
+  onSubmit,
+  values,
+  children,
+}: PropsWithChildren<{
+  schema: Schema;
+  onSubmit: SubmitHandler<Required<z.infer<Schema>>>;
+  values?: Partial<z.infer<Schema>>;
+}>) => {
+  const { handleSubmit, formState, ...restForm } = useForm({
+    resolver: zodResolver(schema),
+    values,
+    resetOptions: {
+      keepDirtyValues: true,
+    },
+  });
+  useSearchFormParams(schema, restForm);
+
+  return (
+    <FormProvider
+      formState={formState}
+      handleSubmit={handleSubmit}
+      {...restForm}
+    >
+      <form onSubmit={handleSubmit(onSubmit)}>{children}</form>
+    </FormProvider>
+  );
+};
+
+const SubmitButton = ({ label }: { label: string }) => {
+  const { formState } = useFormContext();
+  return (
+    <OsdsButton
+      slot="actions"
+      color={ODS_THEME_COLOR_INTENT.primary}
+      {...(formState.isValid ? {} : { disabled: true })}
```

Comment:
```suggestion
      {...(formState.isValid && { disabled: true })}
```

Reply:
It doesn't work with ODS, it requires it or else it would not update